docs/src/pages/ComponentArchitecture.mdx

```diff
@@ -0,0 +1,97 @@
+export const meta = {
+  title: 'Component Architecture',
+}
+
+The design and architecture of components should follow from a set of first principles.
+
+- There is 1 right way to do things, and it's easy to do the right thing.
+- Consumers should not be penalized for code they don't use (through size, perf, etc).
+- Don't overgeneralize until we have a clear need (abstracting should solve a bundle size goal, or DRY goal, or partner use case)
+
+# Base components
+
+Each component should be built upon a "base component".
+
+In order to keep sorted lists formatted neatly, each base component should be named with the form: `ComponentNameBase.tsx`
+
```

Comment:
#### Components should be composable

Composition should allow to apply changes to accessibility, styles or state of components.

```tsx
<Image /> {/* <img src="" /> */}
<RemovedImage /> {/* <img alt="This image was removed" /> */}
```

#### There should be one JSX element in the tree per rendered component

```tsx
// 👍 this is fine
<AvatarImage />
// 👎 bloats React tree
<AvatarImage>
  <Image />
</AvatarImage>
```

Reply:
Suggested update:

```diff
-Composition should allow to apply changes to accessibility, styles or state of components.
+Composition should allow to change accessibility, styles, state, or slots of a components.
```